### DISCUSSION TITLE
Advice for quick bringup on bluerov2-heavy system?

### DISCUSSION THREAD

**Discussion by abubake** [Robotic-Decision-Making-Lab/blue#161] [Q&A]
Hi Evan,

Wondering about the steps necessary to go from an existing BlueROV2/ Heavy system to your system.

For example, is it as simple as: Take the R-PI in the bluerov as is, install everything necessary per your tutorials, then deploy? Does the flight controller matter? For example, older Bluerov's use the pixhawk instead of bluerobotic's navigator flight controller, does that make a difference?

Right now I have a newer heavy config bluerov, and a modified older bluerov that my lab has been slowly building up (adding structures, writing our own ros2 code). I think now is a good time to try out your system but I am trying to gauge which of our subs it would be faster to get it running on. The modified bluerov is running bare bones ubuntu 22 with our ROS2 code for thruster mapping, thrusters are in same physical config as bluerov2 (6 thruster) and we removed the pixhawk and drive the thrusters with a pwm driver on a custom PCB.

Any recommendations would be appreciated :)

Unrelated, but your new updates to the documentation looks great! Awesome work.

**Comment by evan-palmer** [comment]
Hey Baker,

That's a good question. Here are the specs for the tests that we did with Blue installed on the BlueROV2:

* Vehicle: BlueROV2 Heavy (modified to use the Alpha arm)
* Compute: RPi 4 with the Navigator Pi hat
* OS: arm64 version of Raspbian Bullseye
* Installation: Blue was installed onboard the BlueROV2 RPi 4 using Docker. We also did some successful tests with Blue installed on a topside computer, but decided to run things on the vehicle itself.

Here are the steps that we took to get that configuration working:

1. Install a clean version of Raspbian Bullseye 64-bit on the BlueROV2 RPi 4. We used this instead of the off-the-shelf 32-bit RPi image that Blue Robotics provides to avoid any issues with installing the 32-bit version of ROS 2.
2. Install the arm64 BlueOS image (instructions are available [here](https://github.com/bluerobotics/BlueOS/tree/master/install))
3. Download the Blue `*-robot` Docker image onto the BlueROV2 RPi; we did our tests using Iron
4. Setup a new MAVLink endpoint in BlueOS. [Here](https://discuss.bluerobotics.com/t/mavros-ros-2-in-docker-w-navigator/14855) is a relevant thread to help set that up
5. Modify the MAVROS configuration file to use your new MAVLink endpoint (also described [here](https://discuss.bluerobotics.com/t/mavros-ros-2-in-docker-w-navigator/14855))

If you are planning to make changes on-the-fly while doing testing, you can also clone Blue onto the BlueROV2 RPi and then mount the clone into the Blue Docker image. That is what I did while doing debugging to avoid needing to rebuild the image each time I made changes. The VSCode [Remote Development extension](https://code.visualstudio.com/docs/remote/remote-overview) works quite nicely for that too.

I have not tested anything with the Pixhawk. However, you *should* still be able to get Blue to work by running it on a companion computer. I won't make any promises there though. I've used MAVROS successfully on a companion computer with the Pixhawk on UAVs in the past, and the configuration is hypothetically similar.

**Comment by abubake** [comment]
Hi Evan,

Thank you! That is very helpful. In short, I think I will try Blue out on
our Heavy config that has the navigator then. Right now I need a good
system that will enable me to move to waypoints and collect images at
different waypoints in our testing tank. I saw you have a Qualysis system,
we also have one in our lab at UF.

Why did you end up deciding to use blue onboard rather than the topside
computer? also, I had also had some annoyances with the 32-bit version of
ROS2. Much prefer just using 64 bit.

Best,
Baker

**Reply by evan-palmer** [reply]
We integrated a Reach Alpha arm onto our BlueROV2 to do some work on manipulation and were running into limitations with the tether's bandwidth when trying to run Blue + the Alpha + whole body control from the topside computer. We also saw quite a bit of latency when trying to run the Alpha from a topside computer. Unless you happen to need high rate control capabilities, are running tetherless, or have a lot of data running on the tether, you probably don't need to run Blue onboard. @rakeshv24 runs all of his work (MPC) from the topside without any issues

**Comment by evan-palmer** [comment]
@amarburg you might find this thread relevant to our offline discussions
